### ISSUE TITLE
Validation loss not decreasing

### ISSUE THREAD

**Issue by Ziiweb** [NimaVahdat/FoodSeg_mask2former#3]
Hi, 

Thanks for sharing your work.

The validation loss is not decreasing in my case. Is that what should be expected? The only change I did was changing the batch_size from 8 to 2 to run it on a 8GB GPU.

This is my output at the moment:

python -m scripts.run_training
{0: 'background', 1: 'candy', 2: 'egg tart', 3: 'french fries', 4: 'chocolate', 5: 'biscuit', 6: 'popcorn', 7: 'pudding', 8: 'ice cream', 9: 'cheese butter', 10: 'cake', 11: 'wine', 12: 'milkshake', 13: 'coffee', 14: 'juice', 15: 'milk', 16: 'tea', 17: 'almond', 18: 'red beans', 19: 'cashew', 20: 'dried cranberries', 21: 'soy', 22: 'walnut', 23: 'peanut', 24: 'egg', 25: 'apple', 26: 'date', 27: 'apricot', 28: 'avocado', 29: 'banana', 30: 'strawberry', 31: 'cherry', 32: 'blueberry', 33: 'raspberry', 34: 'mango', 35: 'olives', 36: 'peach', 37: 'lemon', 38: 'pear', 39: 'fig', 40: 'pineapple', 41: 'grape', 42: 'kiwi', 43: 'melon', 44: 'orange', 45: 'watermelon', 46: 'steak', 47: 'pork', 48: 'chicken duck', 49: 'sausage', 50: 'fried meat', 51: 'lamb', 52: 'sauce', 53: 'crab', 54: 'fish', 55: 'shellfish', 56: 'shrimp', 57: 'soup', 58: 'bread', 59: 'corn', 60: 'hamburg', 61: 'pizza', 62: ' hanamaki baozi', 63: 'wonton dumplings', 64: 'pasta', 65: 'noodles', 66: 'rice', 67: 'pie', 68: 'tofu', 69: 'eggplant', 70: 'potato', 71: 'garlic', 72: 'cauliflower', 73: 'tomato', 74: 'kelp', 75: 'seaweed', 76: 'spring onion', 77: 'rape', 78: 'ginger', 79: 'okra', 80: 'lettuce', 81: 'pumpkin', 82: 'cucumber', 83: 'white radish', 84: 'carrot', 85: 'asparagus', 86: 'bamboo shoots', 87: 'broccoli', 88: 'celery stick', 89: 'cilantro mint', 90: 'snow peas', 91: ' cabbage', 92: 'bean sprouts', 93: 'onion', 94: 'pepper', 95: 'green beans', 96: 'French beans', 97: 'king oyster mushroom', 98: 'shiitake', 99: 'enoki mushroom', 100: 'oyster mushroom', 101: 'white button mushroom', 102: 'salad', 103: 'other ingredients'}
Some weights of Mask2FormerForUniversalSegmentation were not initialized from the model checkpoint at facebook/mask2former-swin-small-ade-semantic and are newly initialized because the shapes did not match:
- class_predictor.bias: found shape torch.Size([151]) in the checkpoint and torch.Size([105]) in the model instantiated
- class_predictor.weight: found shape torch.Size([151, 256]) in the checkpoint and torch.Size([105, 256]) in the model instantiated
- criterion.empty_weight: found shape torch.Size([151]) in the checkpoint and torch.Size([105]) in the model instantiated
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
/home/tirengarfio/Downloads/FoodSeg_mask2former/FoodSeg_mask2former/lib/python3.12/site-packages/torch/optim/lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:52<00:00,  1.99it/s]
Epoch 1 complete. Avg Training Loss: 39.8190
  0%|                                                                                                                                            | 0/1068 [00:00<?, ?it/s]/home/tirengarfio/Downloads/FoodSeg_mask2former/FoodSeg_mask2former/lib/python3.12/site-packages/datasets/features/image.py:347: UserWarning: Downcasting array dtype int64 to int32 to be compatible with 'Pillow'
  warnings.warn(f"Downcasting array dtype {dtype} to {dest_dtype} to be compatible with 'Pillow'")
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:35<00:00,  4.96it/s]
/home/tirengarfio/.cache/huggingface/modules/evaluate_modules/metrics/evaluate-metric--mean_iou/9e450724f21f05592bfb0255fe2fa576df8171fa060d11121d8aecfff0db80d0/mean_iou.py:259: RuntimeWarning: invalid value encountered in divide
  iou = total_area_intersect / total_area_union
/home/tirengarfio/.cache/huggingface/modules/evaluate_modules/metrics/evaluate-metric--mean_iou/9e450724f21f05592bfb0255fe2fa576df8171fa060d11121d8aecfff0db80d0/mean_iou.py:260: RuntimeWarning: invalid value encountered in divide
  acc = total_area_intersect / total_area_label
Mean IoU: 0.14198435222859002
Validation Loss: 30.2094
Saving model at epoch 1
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:51<00:00,  1.99it/s]
Epoch 2 complete. Avg Training Loss: 26.9230
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:35<00:00,  4.96it/s]
Mean IoU: 0.17945747795859784
Validation Loss: 27.6459
Saving model at epoch 2
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:48<00:00,  2.00it/s]
Epoch 3 complete. Avg Training Loss: 23.2185
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:35<00:00,  4.95it/s]
Mean IoU: 0.2443972532850839
Validation Loss: 24.3715
Saving model at epoch 3
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:49<00:00,  1.99it/s]
Epoch 4 complete. Avg Training Loss: 20.3378
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:36<00:00,  4.94it/s]
Mean IoU: 0.2842785497049795
Validation Loss: 23.7913
Saving model at epoch 4
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:50<00:00,  1.99it/s]
Epoch 5 complete. Avg Training Loss: 18.5163
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:35<00:00,  4.95it/s]
Mean IoU: 0.2830606973214112
Validation Loss: 24.9519
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:50<00:00,  1.99it/s]
Epoch 6 complete. Avg Training Loss: 16.8108
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:35<00:00,  4.95it/s]
Mean IoU: 0.3211299534934465
Validation Loss: 23.1924
Saving model at epoch 6
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:50<00:00,  1.99it/s]
Epoch 7 complete. Avg Training Loss: 15.4615
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:36<00:00,  4.94it/s]
Mean IoU: 0.3253794021963097
Validation Loss: 24.1162
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:50<00:00,  1.99it/s]
Epoch 8 complete. Avg Training Loss: 14.4859
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:36<00:00,  4.94it/s]
Mean IoU: 0.31537451642169745
Validation Loss: 23.2997
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:50<00:00,  1.99it/s]
Epoch 9 complete. Avg Training Loss: 13.3139
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:36<00:00,  4.93it/s]
Mean IoU: 0.33602971711866114
Validation Loss: 23.4478
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:51<00:00,  1.99it/s]
Epoch 10 complete. Avg Training Loss: 12.5989
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:36<00:00,  4.94it/s]
Mean IoU: 0.36049555403409783
Validation Loss: 23.7207
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:50<00:00,  1.99it/s]
Epoch 11 complete. Avg Training Loss: 10.1435
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:35<00:00,  4.95it/s]
Mean IoU: 0.3951460953450288
Validation Loss: 22.4550
Saving model at epoch 11
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:49<00:00,  1.99it/s]
Epoch 12 complete. Avg Training Loss: 9.2907
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:36<00:00,  4.94it/s]
Mean IoU: 0.39931215930422176
Validation Loss: 22.9628
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2492/2492 [20:50<00:00,  1.99it/s]
Epoch 13 complete. Avg Training Loss: 8.8476
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1068/1068 [03:36<00:00,  4.94it/s]
Mean IoU: 0.40811969056324365
Validation Loss: 23.3445
